core/payment/src/processor.rs

```diff
@@ -504,8 +504,13 @@ impl PaymentProcessor {
             Ok(confirmation) => PaymentConfirmation { confirmation },
             Err(e) => return Err(VerifyPaymentError::ConfirmationEncoding),
         };
+        //this need to be rewritten
```

Comment:
Same concerns as in my `THIS CODE NEEDS  REFACTOR` comment. If there's something specific to do please create an issue and link it in this comment.